From 28ef878fc113ed9ffa35b5d83a58919bebf46750 Mon Sep 17 00:00:00 2001
From: Arie Brown <abrown@habana.ai>
Date: Mon, 19 Oct 2020 13:43:23 +0300
Subject: [PATCH] [SW-22147] gaudi: pldm: update its to latest

the current linux.its include fields which weren't defined in plds.its
it resolve error occure in post-image-fit.sh.

Change-Id: Icef77a923c6d0e4d1b830a7fd9805256d2611fcf
Signed-off-by: Arie Brown <abrown@habana.ai>
---

diff --git a/board/habanalabs/common/linux-pldm.its b/board/habanalabs/common/linux-pldm.its
index 78d7ce9..a03254e 100644
--- a/board/habanalabs/common/linux-pldm.its
+++ b/board/habanalabs/common/linux-pldm.its
@@ -10,6 +10,8 @@
 	hl-fw-version = "@HL_FW_VERSION@";
 	hl-device = "@HL_DEV@";
 	description = "Simple image with single Linux kernel and FDT blob";
+	hl-sec-key-idx = <@HL_KEY_INDEX@>;
+	hl-sec-version = <@HL_SEC_FW_VERSION@>;
 	#address-cells = <2>;
 
 	images {
@@ -22,6 +24,12 @@
 			compression = "none";
 			load = <0x00000000 0x08080000>;
 			entry = <0x00000000 0x08080000>;
+			hash@1 {
+				algo = "crc32";
+			};
+			hash@2 {
+				algo = "sha1";
+			};
 		};
 		fdt@1 {
 			description = "Flattened Device Tree blob";
@@ -30,6 +38,9 @@
 			arch = "arm64";
 			compression = "none";
 			load = <0x00000000 0x00000000>;
+			hash@1 {
+				algo = "sha1";
+			};
 		};
 		ramdisk@1 {
 			description = "ramdisk";
@@ -40,6 +51,18 @@
 			compression = "none";
 			load = <0x00000000 0x00000000>;
 			entry = <0x00000000 0x00000000>;
+			hash@1 {
+				algo = "sha1";
+			};
+		};
+		pub-key {
+			description = "Public key in DTB format";
+			data = /incbin/("@KEY_DTB@");
+			type = "standalone";
+			compression = "none";
+			hash {
+				algo = "sha256";
+			};
 		};
 	};
 	configurations {
@@ -49,6 +72,11 @@
 			kernel = "kernel@1";
 			ramdisk = "ramdisk@1";
 			fdt = "fdt@1";
+			signature {
+				algo = "sha256,rsa3072";
+				key-name-hint = "hl_rsa3072";
+				sign-images = "kernel", "ramdisk", "fdt";
+			};
 		};
 	};
 };
